internal/clientconn/listener.go

```diff
@@ -289,6 +289,7 @@ func acceptLoop(ctx context.Context, listener net.Listener, wg *sync.WaitGroup,
 			runCtx = pprof.WithLabels(runCtx, pprof.Labels("conn", connID))
 			pprof.SetGoroutineLabels(runCtx)
 
+			withTLS := l.tlsListener != nil
```

Comment:
What if we have a few listeners, and this particular connection wasn't made through TLS?

Reply:
How do you know if the connection was made with TLS? :) What if we have a few listeners - TLS listener, UNIX socket, and TCP listeners altogether? If the connection came from `acceptLoop` created with `l.tlsListener` then it's from TLS, otherwise it's not.

Reply:
Ohh my bad. I see what is wrong now

Reply:
I'm not exactly sure how I can get the type of listener the connection came in through. `netConn.RemoteAddr().Network()` would return `tcp` for a TLS connection since TLS essentially uses an underlying TCP connection

Reply:
To me, it looks like the simplest approach is to add one more param to `acceptLoop`. And then in the place where we call `accpetLoop` with `l.tlsListener` we can pass this param. And then here instead of `l.tlsListener != nil` we can just use this param passed to a function. 

Like this:
```go
func acceptLoop(ctx context.Context, listener net.Listener, wg *sync.WaitGroup, l *Listener, logger *zap.Logger, isTLS bool)
```

and then

```go
acceptLoop(ctx, l.tlsListener, &wg, l, logger, true)
...
acceptLoop(ctx, l.tlsListener, &wg, l, logger, false)
...
acceptLoop(ctx, l.unixListener, &wg, l, logger, false)
```

It looks a but ugly, but unfortunately, I don't see a more beautiful solution. What do you think?

Reply:
Yeah, I guess you're right, the `crypto/tls` package doesn't have any exported type for a tls listener. A type assertion
 would have gotten the job done